nomad/structs/structs.go

```diff
@@ -2414,6 +2421,9 @@ type Plan struct {
 func (p *Plan) AppendUpdate(alloc *Allocation, status, desc string) {
 	newAlloc := new(Allocation)
 	*newAlloc = *alloc
+	if p.Job != nil {
+		newAlloc.Job = nil // Normalize the job
```

Comment:
So in the case where we are stopping a job, `p.Job` will be `nil` and all the allocations we stop will call AppendUpdate and will have the job populated in their alloc. We could change the logic to be

```
if p.Job != nil {
   newAlloc.Job = nil // Normalize the job
} else {
   p.Job = newAlloc.Job
}
```